config/m4/check_tls.m4

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+# See file LICENSE for terms.
+#
+
+AC_DEFUN([CHECK_TLS],[
+    AC_ARG_WITH([tls],
```

Comment:
what if transport is specified in "tls" but forgot/missed --with-<tls>

Reply:
i don't understand. what is the configure cmd line you are talking about ?
If "--with-tls" is not provided then it equals to "all". All that are found under src/components/tls

Reply:
I mean "--with-tls=ucp,nccl,sharp " specified but not  --with-nccl and --with-sharp

Reply:
first param only defines the TLs that are tried to be compiled. if with-tls has nccl nccl will first run CHECK_NCCL from nccl/configure.m4 and if it is not available then tl/nccl is not compiled. bottom line "--with-nccl" is used inside CHECK_NCCL which in turn only performed if nccl is part of required tls list.